### DISCUSSION TITLE
Enable Vercel DNS

### DISCUSSION THREAD

**Discussion by lukashodac** [vercel/community#3196] [Help] [answered]
### Summary

https://vercel.com/docs/concepts/projects/domains/add-a-domain in section Vercel nameservers “click on the Enable Vercel DNS button”.

Is it an API or a CLI for that?

### Steps to Reproduce

<img width="959" alt="vercel" src="https://github.com/vercel/community/assets/52261075/280cb7e6-7c23-4659-8487-186bab0ae16d">

**Comment by amyegan** [comment]
Hi @jinglestaging. I'm not sure I understand what you're asking. Can you clarify for me? I want to be sure I get you the right information to help you accomplish your goal.

**Reply by lukashodac** [reply]
I would like to enable dns remotely via CLI or REST Vercel API. Not manually clicking a button.

**Reply by amyegan** [reply, marked as answer]
There isn't an official public endpoint for that yet, but you could use the same endpoint used by the dashboard UI.
```
curl --request PATCH \
  --url https://api.vercel.com/v6/domains/<DOMAIN> \
  --header 'Authorization: Bearer <TOKEN>' \
  --header 'Content-Type: application/json' \
  --data '{"zone":true}'
```
Just be aware that it is possible the endpoint may change without warning as it isn't part of the publicly documented API.
